stateless-asks/program/src/instruction.rs

```diff
@@ -47,17 +51,69 @@ pub fn accept_offer(
     taker_mint: &Pubkey,
     authority: &Pubkey,
     token_program_id: &Pubkey,
+    is_native: bool,
+    maker_size: u64,
+    taker_size: u64,
+    bump_seed: u8,
+) -> Instruction {
+    let init_data = StatelessOfferInstruction::AcceptOffer {
+        has_metadata: false,
+        maker_size,
+        taker_size,
+        bump_seed,
+    };
+    let data = init_data.try_to_vec().unwrap();
+    let mut accounts = vec![
+        AccountMeta::new_readonly(*maker_wallet, false),
+        AccountMeta::new_readonly(*taker_wallet, true),
+        AccountMeta::new(*maker_src_account, false),
+        AccountMeta::new(*maker_dst_account, false),
+        AccountMeta::new(*taker_src_account, false),
+        AccountMeta::new(*taker_dst_account, false),
+        AccountMeta::new_readonly(*maker_mint, false),
+        AccountMeta::new_readonly(*taker_mint, false),
+        AccountMeta::new_readonly(*authority, false),
+        AccountMeta::new_readonly(*token_program_id, false),
+    ];
+    if is_native {
+        accounts.push(AccountMeta::new_readonly(system_program::id(), false));
+    }
+    Instruction {
+        program_id: *program_id,
+        accounts,
+        data,
+    }
+}
+
+/// Creates an 'initialize' instruction.
+#[allow(clippy::too_many_arguments)]
+pub fn accept_offer_with_metadata(
+    program_id: &Pubkey,
+    maker_wallet: &Pubkey,
+    taker_wallet: &Pubkey,
+    maker_src_account: &Pubkey,
+    maker_dst_account: &Pubkey,
+    taker_src_account: &Pubkey,
+    taker_dst_account: &Pubkey,
+    maker_mint: &Pubkey,
+    taker_mint: &Pubkey,
+    authority: &Pubkey,
+    token_program_id: &Pubkey,
+    metadata: &Pubkey,
+    creators: &[&Pubkey],
+    is_native: bool,
     maker_size: u64,
     taker_size: u64,
     bump_seed: u8,
 ) -> Instruction {
     let init_data = StatelessOfferInstruction::AcceptOffer {
+        has_metadata: true,
```

Comment:
What happens if people use the wrong instruction? For example, the maker's src token is an NFT, but the taker doesn't use `has_metadata`.  Will there be a failure on transfer or will it succeed and screw over the creators?